pkg/provider/azure_loadbalancer_backendpool.go

```diff
@@ -911,3 +912,203 @@ func removeNodeIPAddressesFromBackendPool(
 
 	return changed
 }
+
+type backendPoolTypePodIP struct {
+	*Cloud
+}
+
+func newBackendPoolTypePodIP(c *Cloud) BackendPool {
+	return &backendPoolTypePodIP{c}
+}
+
+func (bpi *backendPoolTypePodIP) CleanupVMSetFromBackendPoolByCondition(_ *network.LoadBalancer, _ *v1.Service, _ []*v1.Node, _ string, _ func(string) bool) (*network.LoadBalancer, error) {
+	return nil, errors.New("CleanupVMSetFromBackendPoolByCondition is not implemented for pod IP backend pool")
+}
+
+func (bpi *backendPoolTypePodIP) EnsureHostsInPool(service *v1.Service, _ []*v1.Node, _, _, clusterName, lbName string, backendPool network.BackendAddressPool) error {
+	isIPv6 := isBackendPoolIPv6(ptr.Deref(backendPool.Name, ""))
+
+	var (
+		changed               bool
+		err                   error
+		podIPsToBeAdded []string
+		endpointsliceName string
+		endpointSliceNames [] string
+		numOfAdd int
+	)
+
+	endpointSliceList, err := bpi.getEndpointSliceListForService(service)
+
+	if err != nil {
+		klog.Errorf("bpi.EnsureHostsInPool: failed to get endpoint slice list for service %q, error: %s", service.Name, err.Error())
+		return err
+	}
+
+	lbBackendPoolName := bpi.getBackendPoolNameForService(service, clusterName, isIPv6)
+
+	/* Remove all addresses from the backend pool and add the addresses from all the 
+	   endpoint-slices pertaining to a service.*/
+	if strings.EqualFold(ptr.Deref(backendPool.Name, ""), lbBackendPoolName) &&
+		backendPool.BackendAddressPoolPropertiesFormat != nil {
+		if backendPool.LoadBalancerBackendAddresses == nil {
```

Comment:
Do we need these checks? `removeNodeIPAddressesFromBackendPool` checks already and `addPodIPAddressesToBackendPool` inits it as well.